### ISSUE TITLE
Is a new release planned anytime soon?

### ISSUE THREAD

**Issue by bitsofinfo** [breser/git2consul#165]
Is a new release planned anytime soon? Great project, numerous outstanding PRs etc

**Comment by eyalzek** [comment]
please cut a new release, my fix was merged back in August and I'm still getting hit by this bug (https://github.com/breser/git2consul/pull/152)

**Comment by nfedyk** [comment]
please consider doing a new release.

**Comment by joeatbayes** [comment]
Working on a version in GO that handles variable interpolation and inheritance to avoid having thousands of config strings that vary only a small amount  or vary in predictable patterns redefined for every environment.   Suggestions welcome https://github.com/joeatbayes/file2consul   file2consul-dumb.go which handles variable interpolation is working now.  Inheritance will be soon.

**Comment by bitsofinfo** [comment]
@joeatbayes awesome will take a look, so yours will integrate with git as well? would it be a straight migration from git2consul to use yours or a totally different model?

**Comment by ahmedyarub** [comment]
I've tested file2consul and it's a job well done.
However, it is still too basic for most of my needs. I think that somebody should fork git2consul and continue developing it :(

**Comment by shamil** [comment]
what about [gonsul](https://github.com/miniclip/gonsul) looks like a good alternative, but seems also not maintained. Maybe forking it makes sense...

**Comment by joeatbayes** [comment]
> I've tested file2consul and it's a job well done.
> However, it is still too basic for most of my needs. I think that somebody should fork git2consul and continue developing it :(

Please file requests for enhancements with [file2consul ](https://github.com/joeatbayes/file2consul) in it's repository and I will be happy to consider adding them.    It is hard to respond to generic comments like too primitive.

**Comment by joeatbayes** [comment]
> @joeatbayes awesome will take a look, so yours will integrate with git as well? would it be a straight migration from git2consul to use yours or a totally different model?

Yes [file2consul](https://github.com/joeatbayes/file2consul) works from files easily updated using a git pull.    It is pretty trivial migration for basic use but you do not receive some of the more important benefits without a little extra work.  [File2consul](https://github.com/joeatbayes/file2consul) was written to address what I felt were problems in the git2consul design.   file2consul was written to absolutely minimize the number of discrete items that must be edited between environments.      To take advantage of these features you need to change how the properties are saved on disk.  
 file2consul  adds specific support for variable inheritance plus interpolation to reduce maintenance cost for properties when supporting many environments.    To use these advanced features you would need to change how the configuration settings are saved on disk.    

file2consul uses an assertion that different environments are better modeled in separate directories because the branch approach encouraged by git2consul discourages use of inheritance with conventions and interpolation to minimize the total set of manually maintained configuration settings between environments.   This represents a little work but it yields substantial maintenance benefits.

file2consul uses a different assertion for incremental updates.  [File2consul](https://github.com/joeatbayes/file2consul) reads the file set assuming that you have already done a git clone.   It is trivial to run a script that does a git clone in a shell script periodically.  [File2consul](https://github.com/joeatbayes/file2consul) includes a feature that will compare prior run to last time ran so it can update values incrementally as needed.   Net, Net it is a different approach that works just as well.  I made this as an explicit choice because the automatic push of values to consul are problematic when they are not synchronized when they need to be synchronized with software changes.   As such it is better to integrate configuration deployment via the CICD pipeline.

**Comment by hansenc** [comment]
Another alternative is [git2consul-go](https://github.com/KohlsTechnology/git2consul-go) (forked from [go-git2consul](https://github.com/Cimpress-MCP/go-git2consul)) which has some level of atomicity for KV updates.

**Comment by eduardev** [comment]
Hi guys, just found this thread from a friend. Just an heads up as Gonsul is definitely not stalled and never was. It has been used in production heavily on at least my previous 3 major companies (Mainly Miniclip, where it was created at).

I mean, it's just a tool. I created because at the time there was nothing else except this git2consul which was not a 1st choice from a devops perspective. It does the job. There's not much for that tool to do and of course there's always space for more and better. 

Love open source =)
